cmd/nodepool/core/create.go

```diff
@@ -184,3 +191,95 @@ func validateHostedClusterPayloadSupportsNodePoolCPUArch(ctx context.Context, cl
 
 	return nil
 }
+
+// validMinorVersionCompatibility validates that the NodePool version is compatible with the HostedCluster version.
+// For 4.even versions, it allows y-2 difference.
+// For 4.odd versions, it allows y-1 difference.
+// NodePool version cannot be higher than control plane version.
+func validMinorVersionCompatibility(ctx context.Context, client crclient.Client, name, namespace, nodePoolReleaseImage string) error {
+	if nodePoolReleaseImage == "" {
+		return nil
+	}
+	logger := ctrl.LoggerFrom(ctx)
+
+	hcluster := &hyperv1.HostedCluster{}
+	if err := client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: name}, hcluster); err != nil {
```

Comment:
You probably want to distinguish the [`IsNotFound`][1] case from other errors, so you don't treat situations like Kube-API hiccups or RBAC wrinkles as "do whatever you want!" unguarded windows.

[1]: https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsNotFound

Reply:
done